### PR TITLE
Support checkstyle report output #108. Support for the -r and -f para…

### PR DESCRIPTION
**Pull request by ngeor**
…meters of scss-lint.

This enables to set in options

    format: 'Checkstyle',
    require: 'scss_lint_reporter_checkstyle'

The names are in compliance with the CLI tool:

    Usage: scss-lint [options] [scss-files]
        -f, --format Formatter           Specify how to display lints
        -r, --require path               Require Ruby file

As a side-effect, with my fix you can use all built-in formatters of scss-lint (Config, Files, JSON, XML) as long as you use the reporterOutput option of grunt-scss-lint.

### CODE REVIEW

**Comment by mariusbuescher** [comment]
Try setting the `options.timeout` configuration in Gruntfile.js to a higher value, e.g. `2500`. It seems other developers struggle with similar problems.

**Comment by ngeor** [comment]
AppVeyor fails, Travis passes. If I go to AppVeyor history for this project, I can't really find any build that was successful. Is it possible that it is misconfigured?

**Comment by ngeor** [comment]
I've updated the PR. It passes on Travis. On AppVeyor things are more difficult. I had to tell mocha-istanbul to hide the output of mocha + istanbul. AppVeyor gets confused and treats that output as an error. But even with this, AppVeyor dies at the final command, regarding code climate. That one however in the documentation says it only works for the master branch.

In the PR I've done some further changes:

- I've updated all gem and node dependencies to the latest, as the master is quite old.
- I've set AppVeyor to run also for node 4.4 (Travis picks up the latest anyway)
- Unit tests are written in various styles and most of them are integration tests anyway. I grouped them based on the technique that is being used.

At this point, I can't really proceed any further unless @ahmednuaman has a look at the PR and my findings regarding AppVeyor.

Also, I bumped up the version from 0.3.8 to 0.4.0 but perhaps it is worth bumping it up to 1.0.0 since all dependencies are upgraded.

**Comment by ahmednuaman** [comment]
Let me remove this appveyor nonsense.

**Comment by ahmednuaman** [comment]
Can you squash the commits please and hopefully that'll kick off the tests again.

**Comment by ngeor** [comment]
Hm I don't see travis picking it up. I even added a dummy commit in .gitignore

**Comment by ngeor** [comment]
Actually it works: https://travis-ci.org/ahmednuaman/grunt-scss-lint/builds/142527737 I don't know why it's not displayed inline in the PR in GitHub

**Comment by ahmednuaman** [comment]
Can you rebase please?

**Comment by ngeor** [comment]
Of course, should be fine now! Here's the latest build: https://travis-ci.org/ahmednuaman/grunt-scss-lint/builds/142999385

File: Gemfile.lock [hunk 1]
```diff
@@ -1,14 +1,20 @@
 GEM
   remote: https://rubygems.org/
   specs:
-    rainbow (2.0.0)
-    sass (3.4.16)
-    scss_lint (0.40.1)
-      rainbow (~> 2.0)
-      sass (~> 3.4.1)
+    rake (11.1.2)
```
Comment (ahmednuaman) [inline Gemfile.lock]: Why rake?

Comment (ngeor) [inline Gemfile.lock]: It gets added when I run 'bundle'. Could be because of the upgrade I did on the Gemfile?

Comment (ahmednuaman) [inline Gemfile.lock]: I'm not sure if it's needed cos rake tasks aren't used here.

**Comment by ngeor** [comment]
I took out the 'rake' dependency from Gemfile.lock . Latest build: https://travis-ci.org/ahmednuaman/grunt-scss-lint/builds/143670110

It fails with message: "Your Gemfile.lock is corrupt. The following gem is missing from the DEPENDENCIES
section: 'rake'"

**Comment by ngeor** [comment]
I also removed the 'rake' nested dependency underneath scss-lint, same result. Like I said initially, I have not added the 'rake' dependency myself. In Gemfile, it is not present. This is apparently a dependency of scss-lint and it gets written into the generated file Gemfile.lock once one runs 'bundle'.

**Comment by ngeor** [comment]
Thanks for merging!

**Pull request merged**